src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period ");
+        result.append(getFormattedData(dateFrom, "-", dateTo));
+        for (String name : names) {
+            int sumSalary = 0;
+            for (String value : data) {
+                String[] dataSplitted = value.split(" ");
+                if (isWithinPeriod(dataSplitted[0], dateFrom, dateTo)
+                        && name.equals(dataSplitted[1])) {
+                    sumSalary += Integer.parseInt(dataSplitted[2])
+                            * Integer.parseInt(dataSplitted[3]);
+                }
```

Comment:
Avoid using nested loops if possible. Since you are iterating through all the data for each name, consider refactoring to reduce the number of loops and improve efficiency.